clang/lib/Lex/PPDirectives.cpp

```diff
@@ -1073,6 +1076,74 @@ OptionalFileEntryRef Preprocessor::LookupFile(
   return std::nullopt;
 }
 
+OptionalFileEntryRef
```

Comment:
Is the search here different enough from how `include` does its lookup to require its own complete implementation?

Reply:
I think they are different enough to justify different interfaces (include files are resolved relative to the current inclusion stack, has to handle subframeworks, there's different search options, etc.)

Reply:
I guess I think its probably ok to have different interfaces, I'm just concerned about how much 'new' code there is here for something we already do 95% of.  That is, should/could the two bits of functionality share an impl function?  Or are they that dissimilar that it isn't valuable?

Reply:
I don't think it's 95% the way things are currently structured (include search has different paths than embed search, has to handle frameworks, has a different notion of "system" paths, etc). I think we could get some code sharing in here at some point, but I think by the time we finish extracting the common bits, we'll wind up with about the same amount of code.